### PR TITLE
DevOps: Fix json query in reading the docker names to filter out fields not starting with aiida

### PR DESCRIPTION
**Pull request by agoscinski**
The BAKE_METADATA ha another field with warning information (`buildx.build.warnings`) that does not contain `image.name` so the json query failed (. Now I added a json query filtering out every field name that does not start with "aiida".
[bake_metadata.json](https://github.com/user-attachments/files/17161429/bake_metadata.json)

### CODE REVIEW

**Comment by danielhollas** [comment]
As a quick short term fix you might want to pin to docker/bake-action to v5.5.0
This is what we did in aiidalab-docker-stack, see https://github.com/aiidalab/aiidalab-docker-stack/issues/493

I am just on my way to holiday and will be out for two weeks so don't have time to look into this more. @unkcpz can review this instead perhaps.

File: .github/workflows/extract-docker-image-names.sh [hunk 1]
```diff
@@ -52,5 +49,7 @@ if [[ -z ${BAKE_METADATA-} ]];then
     exit 1
 fi
 
-images=$(echo "${BAKE_METADATA}" | jq -c '. as $base |[to_entries[] |{"key": (.key|ascii_upcase|sub("-"; "_"; "g") + "_IMAGE"), "value": [(.value."image.name"|split(",")[0]),.value."containerimage.digest"]|join("@")}] |from_entries')
+images=$(echo "${BAKE_METADATA}" |
+jq -c 'to_entries | map(select(.key | startswith("aiida"))) | from_entries' | # filters out every key that does not start with aiida
```
[the remark this answers is not in the document]
Comment (agoscinski) [reply .github/workflows/extract-docker-image-names.sh:53]: Yes agree, understanding this json query was very cumbersome. I would prefer some Python code for this repo.

[the remark this answers is not in the document]
Comment (danielhollas) [reply .github/workflows/extract-docker-image-names.sh:53]: Agreed, if you can figure out some Python one-liner that would be best.

**Comment by unkcpz** [comment]
Why the CI test is not triggered?

**Comment by agoscinski** [comment]
> Why the CI test is not triggered?

Because it only happens if you push to aiida-core with a branch that fulfills 
> Matches all branch and tag names that don't contain a slash (/). The * character is a special character in YAML. When you start a pattern with *, you must use quotes.

https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#patterns-to-match-branches-and-tags
See workflow file https://github.com/aiidateam/aiida-core/blob/c7c289d3892bf76894714f53f58b7ce5b0761178/.github/workflows/docker.yml#L7-L9

File: .github/workflows/docker-build-test.yml [hunk 2]
```diff
@@ -41,7 +41,7 @@ jobs:
       uses: docker/setup-buildx-action@v3
 
     - name: Build images
-      uses: docker/bake-action@v5
+      uses: docker/bake-action@v5.5.0
```
Comment (unkcpz) [inline .github/workflows/docker-build-test.yml:44]:
```suggestion
      uses: docker/bake-action@v5
```

File: .github/workflows/docker-build.yml [hunk 3]
```diff
@@ -54,7 +54,7 @@ jobs:
 
     - name: Build and upload to ghcr.io 📤
       id: build
-      uses: docker/bake-action@v5
+      uses: docker/bake-action@v5.5.0
```
Comment (unkcpz) [inline .github/workflows/docker-build.yml:57]:
```suggestion
      uses: docker/bake-action@v5
```

**Review by unkcpz — APPROVED** [review]
Looks good to me.

**Pull request merged**